api/common.js

```diff
@@ -491,6 +488,67 @@ exports.updateProjectMag = function(project, cb) {
     });
 }
 
+exports.updatePublicationMag = function(publication,cb){
+    console.log("--- %s %s", publication.name, publication._id.toString());
+    if (!config.mag) cb("no mag config!!");
+    if(!publication.mag) publication.mag = {}; //not sure if we need this or not
+    publication.markModified("mag");
+
+    const query = exports.generateQuery(publication.name+" "+publication.desc+" "+publication.readme);
+
+    if(!query){
+        publication.mag.papers = [];
+        publication.save(cb);
+        return;
+    }
+    console.debug(query);
+    axios.get("https://api.labs.cognitive.microsoft.com/academic/v1.0/evaluate", { 
```

Comment:
this block is very similar to the one for project, right? Should we refactor this?

Reply:
create a function for it